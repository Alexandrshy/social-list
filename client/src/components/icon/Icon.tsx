import React, { memo } from "react";

import {
  SIZE_RESPONSIVE,
  SIZE_SMALL,
  SIZE_MEDIUM,
  SIZE_LARGE,
  TYPE_USE,
  TYPE_IMAGE
} from "./constants";
import * as commonIcon from "./svg/common";

import s from "./Icon.module.css";

const ICONS: { [key: string]: any } = {
  ...commonIcon
};

const SIZES: { [key: string]: string | number } = {
  [SIZE_SMALL]: 16,
  [SIZE_MEDIUM]: 22,
  [SIZE_LARGE]: 50,
  [SIZE_RESPONSIVE]: "100%"
};

type PropsType = {
  name: string;
  size?: string;
  type?: string;
};

export const Icon: React.FC<PropsType> = ({
  name,
  size = SIZE_RESPONSIVE,
  type = "image"
}) => {
  const icon = ICONS[name];
  const sizeView = SIZES[size];

  if (typeof icon === "undefined") return null;

  return (
    <svg
      width={sizeView}
      height={sizeView}
      viewBox={icon.viewBox}
      className={s.container}
    >
      {type === TYPE_USE && <use xlinkHref={`${icon}#${name}`} />}
      {type === TYPE_IMAGE && (
        <image xlinkHref={icon} width={sizeView} height={sizeView} />
      )}
    </svg>
  );
};
