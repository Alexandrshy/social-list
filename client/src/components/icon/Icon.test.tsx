import React from "react";
import renderer from "react-test-renderer";

import { Icon } from "./Icon";

describe("Icon component", () => {
  it("should render an image", () => {
    const tree = renderer.create(<Icon name="logo" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render an use", () => {
    const tree = renderer.create(<Icon name="logo" type="use" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should return null", () => {
    const tree = renderer.create(<Icon name="cats" type="use" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
