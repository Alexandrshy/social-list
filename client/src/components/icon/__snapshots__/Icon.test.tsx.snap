// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should render an image 1`] = `
<svg
  className="container"
  height="100%"
  width="100%"
>
  <image
    height="100%"
    width="100%"
    xlinkHref="logo.svg"
  />
</svg>
`;

exports[`Icon component should render an use 1`] = `
<svg
  className="container"
  height="100%"
  width="100%"
>
  <use
    xlinkHref="logo.svg#logo"
  />
</svg>
`;

exports[`Icon component should return null 1`] = `null`;
